mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -591,28 +623,72 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     if (labelType == 0) {
       // Regression
       ensembleType match {
-        case "RandomForest" =>
-          Left(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
-            "variance", treeDepth, maxBins, this.getRandomSeed))
-        case "GradientBoostedTrees" =>
+        case "mllib.RandomForest" =>
+          MLlibRFModel(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees,
+            featureSubsetStrategy, "variance", treeDepth, maxBins, this.getRandomSeed))
+        case "ml.RandomForest" =>
+          val sqlContext = new SQLContext(rdd.sparkContext)
+          val dataset = sqlContext.createDataFrame(rdd)
+          val model = new RandomForestRegressor()
+            .setImpurity("variance")
+            .setMaxDepth(treeDepth)
+            .setMaxBins(maxBins)
+            .setNumTrees(numTrees)
+            .setFeatureSubsetStrategy(featureSubsetStrategy)
+            .setSeed(this.getRandomSeed)
+            .fit(dataset)
+          MLRFRegressionModel(model)
+        case "mllib.GradientBoostedTrees" =>
           val treeStrategy = new Strategy(Algo.Regression, Variance, treeDepth,
             labelType, maxBins, QuantileStrategy.Sort, categoricalFeaturesInfo)
-          val boostingStrategy = BoostingStrategy(treeStrategy, SquaredError, numIterations = 100,
+          val boostingStrategy = BoostingStrategy(treeStrategy, SquaredError, numTrees,
             learningRate = 0.1)
-          Right(GradientBoostedTrees.train(rdd, boostingStrategy))
+          MLlibGBTModel(GradientBoostedTrees.train(rdd, boostingStrategy))
+        case "ml.GradientBoostedTrees" =>
+          val sqlContext = new SQLContext(rdd.sparkContext)
+          val dataset = sqlContext.createDataFrame(rdd)
+          val model = new GBTRegressor()
```

Comment:
The only loss available for regression is L2 (for both mllib and ml); logistic is only for classification.  My comment was just a precaution in case we add a new loss for regression later on.